File: ant-sdk/antd-cpp/examples/03-chunks.cpp
#include <iostream>
#include <string>
#include <vector>
#include "antd/antd.hpp"

int main() {
    try {
        antd::Client client;

        // Store a raw chunk
        std::string raw = "raw chunk payload";
        std::vector<uint8_t> data(raw.begin(), raw.end());

        auto result = client.chunk_put(data);
        std::cout << "Chunk stored at: " << result.address << "\n";
        std::cout << "Cost: " << result.cost << " atto\n";

        // Retrieve the chunk
        auto chunk = client.chunk_get(result.address);
        std::string text(chunk.begin(), chunk.end());
        std::cout << "Retrieved chunk: " << text << "\n";

    } catch (const antd::AntdError& e) {
        std::cerr << "Error: " << e.what() << "\n";
        return 1;
    }
    return 0;
}

File: ant-sdk/antd-lua/examples/03-chunks.lua
--- Example: Store and retrieve raw chunks.

local antd = require("antd")

local client = antd.new_client()

-- Store a chunk
local chunk_data = "Raw chunk content"
local result, err = client:chunk_put(chunk_data)
if err then
    print("Chunk put error: " .. err.message)
    os.exit(1)
end

print("Chunk stored at: " .. result.address)
print("Cost: " .. result.cost .. " atto")

-- Retrieve the chunk
local retrieved, err2 = client:chunk_get(result.address)
if err2 then
    print("Chunk get error: " .. err2.message)
    os.exit(1)
end

print("Retrieved chunk: " .. retrieved)
